appinventor/buildserver/src/com/google/appinventor/buildserver/Compiler.java

```diff
@@ -2150,14 +2150,12 @@ private boolean insertNativeLibs(File buildDir){
           }
 
           String sourcePath = "";
-          String pathSuffix = RUNTIME_FILES_DIR + sourceDirName + ZIPSLASH + lib;
-
           if (simpleCompTypes.contains(type)) {
+            final String pathSuffix = RUNTIME_FILES_DIR + sourceDirName + ZIPSLASH + lib;
             sourcePath = getResource(pathSuffix);
           } else if (extCompTypes.contains(type)) {
+            final String pathSuffix = "/libsNatives/" + sourceDirName + ZIPSLASH + lib;
             sourcePath = getExtCompDirPath(type) + pathSuffix;
-            targetDir = createDir(targetDir, EXT_COMPS_DIR_NAME);
```

Comment:
I did this because I tried to use an external library together with the natives and I couldn't. Because the native was not at the root (x86_64 / lib.so), it was at (x86_64 / package / lib.so).

Reply:
Hello @Evan, I just analyzed it better and here's the answer. With this directory structure, when calling `apkBuilder.addNativeLibraries (libsDir);` the natives of the external components are not added, because it is in that structure `/folder/folder/folder/file`, while `.addNativeLibraries` compiles in this way `/folder/folder/file`. Check here: https://android.googlesource.com/platform/sdk/+/e162064a7b5db1eecec34271bc7e2a4296181ea6/sdkmanager/libs/sdklib/src/com/android/sdklib/build/ApkBuilder.java#619

Reply:
@ewpatton